src/fsearch_list_view.c

```diff
@@ -1110,6 +1123,49 @@ on_fsearch_list_view_bin_drag_gesture_begin(GtkGestureDrag *gesture,
     }
 }
 
+static void
+on_drag_data_get(GtkWidget *widget,
+                 GdkDragContext *context,
+                 GtkSelectionData *selection_data,
+                 guint info,
+                 guint32 time,
+                 gpointer user_data) {
+    FsearchListView *view = FSEARCH_LIST_VIEW(widget);
+    if (!view->drag_data_get_func) {
+        g_warning("drag_data_get_func is not set for FsearchListView.");
+        return;
+    }
+
+    GList *indices = NULL;
+    for (int i = 0; i < view->num_rows; i++) {
+        if (view->is_selected_func(i, view->selection_user_data)) {
+            indices = g_list_append(indices, GINT_TO_POINTER(i));
+        }
+    }
+
+    GList *uris = view->drag_data_get_func(indices, view->selection_user_data);
+
+    g_list_free(indices);
+
+    if (uris) {
+        gchar **uri_array = g_new(gchar *, g_list_length(uris) + 1);
+        int index = 0;
+        for (GList *l = uris; l != NULL; l = l->next) {
+            uri_array[index++] = g_strdup((gchar *)l->data);
+        }
+        uri_array[index] = NULL;
+        gtk_selection_data_set_uris(selection_data, uri_array);
+        g_list_free_full(uris, g_free);
+        for (int i = 0; uri_array[i] != NULL; i++) {
```

Comment:
you can use `g_clear_pointer(&uri_array, g_strfreev);` here instead to clean up the whole string array.